### ISSUE TITLE
Distributed.RemoteChannel & DataFrames

### ISSUE THREAD

**Issue by dwfmarchant** [JuliaData/DataFrames.jl#1679]
I'm having a problem working with Distributed.RemoteChannel any time DataFrames is loaded.  I've come up with this minimal example to reproduce the problem:

```
using DataFrames
using Distributed
addprocs(1)
out = RemoteChannel(()->Channel(1), 2)
println(out)
```

on mac, this results in:
```
ERROR: LoadError: StackOverflowError:
deserialize(::Distributed.ClusterSerializer{Sockets.TCPSocket}, ::Type{RemoteChannel{Channel{Any}}}) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Distributed/src/remotecall.jl:310 (repeats 100 times)
Stacktrace:
 [1] #remotecall_fetch#149(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Function, ::Distributed.Worker, ::Function, ::Vararg{Any,N} where N) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Distributed/src/remotecall.jl:379
 [2] remotecall_fetch(::Function, ::Distributed.Worker, ::Function, ::Vararg{Any,N} where N) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Distributed/src/remotecall.jl:371
 [3] #remotecall_fetch#152(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Function, ::Int64, ::Function, ::Vararg{Any,N} where N) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Distributed/src/remotecall.jl:406
 [4] remotecall_fetch at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Distributed/src/remotecall.jl:406 [inlined]
 [5] RemoteChannel(::Function, ::Int64) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Distributed/src/remotecall.jl:108
 [6] top-level scope at none:0
 [7] include at ./boot.jl:317 [inlined]
 [8] include_relative(::Module, ::String) at ./loading.jl:1044
 [9] include(::Module, ::String) at ./sysimg.jl:29
 [10] exec_options(::Base.JLOptions) at ./client.jl:266
 [11] _start() at ./client.jl:425
```

Removing the `using DataFrames` runs no problem.  The way it is crashes seems to vary depending on the platform I'm running it on.  On linux (Ubuntu 16.04.4) it does not crash, but hangs on the RemoteChannel line while slowly ramping memory use to 100%.

**Comment by dwfmarchant** [comment]
Actually, it looks like this might be a julia bug (JuliaLang/julia#30679).

**Comment by bkamins** [comment]
@pszufe - can you have a look at it?

**Comment by dwfmarchant** [comment]
This is fixed by [JuliaLang/julia/pull/30880](https://github.com/JuliaLang/julia/pull/30880).
